baseapp/baseapp_test.go

```diff
@@ -105,6 +110,64 @@ func setupBaseApp(t *testing.T, options ...func(*BaseApp)) *BaseApp {
 	return app
 }
 
+// simple one store baseapp with data and snapshots. Each tx is 1 MB in size (uncompressed).
+func setupBaseAppWithSnapshots(t *testing.T, blocks uint, blockTxs int, options ...func(*BaseApp)) (*BaseApp, func()) {
+	codec := codec.New()
+	registerTestCodec(codec)
+	routerOpt := func(bapp *BaseApp) {
+		bapp.Router().AddRoute(sdk.NewRoute(routeMsgKeyValue, func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+			kv := msg.(*msgKeyValue)
+			bapp.cms.GetCommitKVStore(capKey2).Set(kv.Key, kv.Value)
+			return &sdk.Result{}, nil
+		}))
+	}
+
+	snapshotDir, err := ioutil.TempDir("", "baseapp")
+	require.NoError(t, err)
+	snapshotStore, err := snapshots.NewStore(dbm.NewMemDB(), snapshotDir)
+	require.NoError(t, err)
+	teardown := func() {
+		os.RemoveAll(snapshotDir)
+	}
+
+	app := setupBaseApp(t, append(options,
+		SetSnapshotStore(snapshotStore),
+		SetSnapshotInterval(2),
```

Comment:
This is a test fixture. If we use a larger number (e.g. 10) then the tests will take much longer, since we need to wait for additional blocks to be produced before we get the necessary number of snapshots needed by the test.